src/UserService.Validation/User/EditUserRequestValidator.cs

```diff
@@ -1,114 +1,206 @@
+                    { x => NameRegex.IsMatch(x.value.ToString()), "FirstName has incorrect format" }
+                });
+
+            #endregion
+
+            #region lastname
+
+            AddFailureForPropertyIf(
+                nameof(EditUserRequest.LastName),
+                x => x == OperationType.Replace,
+                new Dictionary<Func<Operation<EditUserRequest>, bool>, string>
+                {
+                    { x => !string.IsNullOrEmpty(x.value.ToString()), "LastName is too short" },
+                    { x => x.value.ToString().Length < 100, "LastName is too long" },
+                    { x => NameRegex.IsMatch(x.value.ToString()), "LastName has incorrect format" }
+                });
+
+            #endregion
+
+            #region middlename
+
+            AddFailureForPropertyIf(
+                nameof(EditUserRequest.MiddleName),
+                x => x == OperationType.Replace || x == OperationType.Add,
+                new Dictionary<Func<Operation<EditUserRequest>, bool>, string>
+                {
+                    { x => !string.IsNullOrEmpty(x.value.ToString()), "MiddleName is too short" },
+                    { x => x.value.ToString().Length < 32, "MiddleName is too long" },
+                    { x => NameRegex.IsMatch(x.value.ToString()), "MiddleName has incorrect format" }
+                });
+
+            #endregion
+
+            #region Status
+
+            AddFailureForPropertyIf(
+                nameof(EditUserRequest.Status),
+                x => x == OperationType.Replace,
+                new Dictionary<Func<Operation<EditUserRequest>, bool>, string>
+                {
+                    { x => Enum.TryParse(typeof(UserStatus), x.value?.ToString(), out _), "Incorrect user status"},
                 });
+
+            #endregion
+
+            #region Rate
+
+            AddFailureForPropertyIf(
+                nameof(EditUserRequest.Rate),
+                x => x == OperationType.Replace,
+                new Dictionary<Func<Operation<EditUserRequest>, bool>, string>
+                {
+                    { x => double.TryParse(x.value?.ToString(), out _), "Incorrect rate format"},
+                    { x =>
+                        {
+                            if (double.TryParse(x.value?.ToString(), out double rate))
+                            {
+                                return rate > 0;
+                            }
+
+                            return false;
+                        },
+                        "Rate must be greater than 0"
+                    },
+                    { x =>
+                        {
+                            if (double.TryParse(x.value?.ToString(), out double rate))
+                            {
+                                return rate <= 1;
+                            }
+
+                            return false;
+                        },
+                        "Rate must be less than 3"
```

Comment:
Rate must be less or equal than 1, I think

Reply:
fixed